### ISSUE TITLE
Cookie sameSite setting

### ISSUE THREAD

**Issue by manu37** [cmsimple-xh/cmsimple-xh#451]
https://blog.chromium.org/2019/10/developers-get-ready-for-new.html

**Comment by cmb69** [comment]
I don't think that XH core is affected by this.

**Comment by manu37** [comment]
"Das Cookie “XH_2f434d53696d706c654769742f” wird bald als Cross-Site-Cookie gegen “http://localhost/cmsimplegit/?Start” behandelt, da das Schema nicht übereinstimmt. cmsimplegit"

**Comment by cmb69** [comment]
Aber "cmsimplegit" ist doch nur Teil des Pfads, nicht des Schemas und nicht der Domain? Oder stören sich Browser bald daran, dass das Cookie für den Root-Pfad (`/`) gesetzt wird? Jedenfalls wäre es bedauerlich, wenn für dieses Cookie SameSite:None gesetzt werden müsste, weil das die Sicherheit eben nicht verbessern würde, und man local nur noch per HTTPS arbeiten könnte.

**Comment by manu37** [comment]
Du hast wohl recht. Bei einer Online Installation wird nichts gemeckert. Zu prüfen wäre aber eine Online Installation in einem Unterverzeichnis.

**Comment by frase-git** [comment]
Online meldet FF:

> Einige Cookies verwenden das empfohlene "SameSite"-Attribut inkorrekt. 3
> - Das Cookie "status" wird in Zukunft bald abgelehnt werden, da es für das Attribut "SameSite" entweder "None" oder einen ungültigen Wert angibt, ohne das "secure"-Attribut zu verwenden. Weitere Informationen zum "SameSite"-Attribut finden Sie unter https://developer.mozilla.org/docs/Web/HTTP/Headers/Set-Cookie/SameSite.
> - Das Cookie "XH_2f78682d646576732d382f" wird in Zukunft bald abgelehnt werden, da es für das Attribut "SameSite" entweder "None" oder einen ungültigen Wert angibt, ohne das "secure"-Attribut zu verwenden. Weitere Informationen zum "SameSite"-Attribut finden Sie unter https://developer.mozilla.org/docs/Web/HTTP/Headers/Set-Cookie/SameSite.
> - Das Cookie "mode" wird in Zukunft bald abgelehnt werden, da es für das Attribut "SameSite" entweder "None" oder einen ungültigen Wert angibt, ohne das "secure"-Attribut zu verwenden. Weitere Informationen zum "SameSite"-Attribut finden Sie unter https://developer.mozilla.org/docs/Web/HTTP/Headers/Set-Cookie/SameSite.

Offline (local) meldet FF:

> - Das Cookie "mode" wird in Zukunft bald abgelehnt werden, da es für das Attribut "SameSite" entweder "None" oder einen ungültigen Wert angibt, ohne das "secure"-Attribut zu verwenden. Weitere Informationen zum "SameSite"-Attribut finden Sie unter https://developer.mozilla.org/docs/Web/HTTP/Headers/Set-Cookie/SameSite. index.php
> - Das Cookie "mode" wurde abgelehnt, da es bereits abgelaufen ist.

**Comment by olape-git** [comment]
https://github.com/cmsimple-xh/cmsimple-xh/blob/e3cc319f5317f6556bd23a68ad61e9209eab68b5/cmsimple/classes/Controller.php#L177
->
`            $arr_cookie_options = array (
                'expires' => 0,
                'path' => CMSIMPLE_ROOT,
                'samesite' => 'Strict'
                );
            setcookie('status', 'adm', $arr_cookie_options);`

https://github.com/cmsimple-xh/cmsimple-xh/blob/e3cc319f5317f6556bd23a68ad61e9209eab68b5/cmsimple/classes/Controller.php#L208
->
`        $arr_cookie_options = array (
            'expires' => 0,
            'path' => CMSIMPLE_ROOT,
            'samesite' => 'Strict'
        );
        setcookie('status', 'adm', $arr_cookie_options);`

https://github.com/cmsimple-xh/cmsimple-xh/blob/e3cc319f5317f6556bd23a68ad61e9209eab68b5/cmsimple/classes/Controller.php#L477
->
`        if (XH_ADM) {
            $arr_cookie_options = array (
                'expires' => 0,
                'path' => CMSIMPLE_ROOT,
                'samesite' => 'Strict'
            );`

https://github.com/cmsimple-xh/cmsimple-xh/blob/e3cc319f5317f6556bd23a68ad61e9209eab68b5/cmsimple/classes/Controller.php#L479
->
`setcookie('mode', 'edit', $arr_cookie_options);`

https://github.com/cmsimple-xh/cmsimple-xh/blob/e3cc319f5317f6556bd23a68ad61e9209eab68b5/cmsimple/classes/Controller.php#L482
->
`setcookie('mode', '', $arr_cookie_options);`

https://github.com/cmsimple-xh/cmsimple-xh/blob/e3cc319f5317f6556bd23a68ad61e9209eab68b5/cmsimple/classes/Controller.php#L489
->
`setcookie('status', '', $arr_cookie_options);`

https://github.com/cmsimple-xh/cmsimple-xh/blob/e3cc319f5317f6556bd23a68ad61e9209eab68b5/cmsimple/classes/Controller.php#L492
->
`setcookie('mode', '', $arr_cookie_options);`

Fehlt noch **XH_2f312e385f73....**

**Comment by cmb69** [comment]
Die erweiterte [setcookie()](https://www.php.net/setcookie) Signatur wird leider erst ab PHP 7.3.0 unterstützt. Ich könnte mir aber vorstellen, dass das Problem erst ab dieser Version auftritt. Wenn das so ist, dann könnten wir per Versionsprüfung die passenden Argumente übergeben. Falls nicht, dann ist ein anderes Übergeben angebracht. ;)

**Comment by cmb69** [comment]
Ich denke, wir *müssen* hier eigentlich nichts machen. Wir setzen kein SameSite-Attribut, und gemäß dem aktuellen Standard, sollte dann der Default "Lax" sein. Das ist im Firefox aber noch nicht der Fall, außer wenn man `network.cookie.sameSite.laxByDefault` aktiviert.

`SameSite=Strict` und `Secure` falls HTTPS, und `HttpOnly` würde wohl sinnvoll sein, aber ich denke es genügt, das nur ab PHP 7.3.0 zu tun (und das entsprechend zu dokumentieren). Für die Sitzung-Cookies könnten wir entsprechend [session_set_cookie_params()](https://www.php.net/manual/en/function.session-set-cookie-params.php) bemühen.

**Comment by frase-git** [comment]
Unter PHP 7.25 im FF gleiche Warnungen in der Console.

**Comment by cmb69** [comment]
> Unter PHP 7.25 im FF gleiche Warnungen in der Console.

Ja, das hat nichts mit PHP, sondern mit Firefox zu tun. Siehe [oben](https://github.com/cmsimple-xh/cmsimple-xh/issues/451#issuecomment-751366288).

**Comment by olape-git** [comment]
> Die erweiterte [setcookie()](https://www.php.net/setcookie) Signatur wird leider erst ab PHP 7.3.0

Da hat Christoph wahr.

Hm, einfach ignorieren? Warten bis FF network.cookie.sameSite.laxByDefault per Standard aktiviert?
